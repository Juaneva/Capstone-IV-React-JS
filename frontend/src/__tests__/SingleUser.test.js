import React from "react";
import { SingleUser } from "../components/SingleUser";
import renderer from "react-test-renderer";
const fetch = require("node-fetch");

//Snapshot test for the single user card:
test("SingleUser is shown", () => {
  const user = {
    id: 68243499,
    name: "Juaneva",
    login: "Juaneva",
    avatar: "https://avatars.githubusercontent.com/u/68243499?v=4",
    url: "https://api.github.com/users/juaneva",
    source: "Github",
  };
  const tree = renderer
    .create(
      <SingleUser
        user={user}
        //Declare empty functions here, as this prop and its type (function)
        //is required by propTypes and therefore the test will fail if not
        //included here
        backToSearch={function backToSearch() {}}
        displayRepo={function displayRepo() {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//Unit test for the fetch that retrieves the single user info:
test("displayUser fetches data correctly", () => {
  return fetch("http://localhost:3001/api/user?source=Github&user=Juaneva")
    .then((data) => data.json())
    .then((data) => {
      expect(data).toEqual({
        id: 68243499,
        name: "Juaneva",
        login: "Juaneva",
        avatar: "https://avatars.githubusercontent.com/u/68243499?v=4",
        url: "https://api.github.com/users/juaneva",
        source: "Github",
      });
    });
});
