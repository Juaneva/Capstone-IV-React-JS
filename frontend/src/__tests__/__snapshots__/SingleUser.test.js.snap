// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUser is shown 1`] = `
<div
  className="single-user"
>
  <h2>
    Juaneva
  </h2>
  <div
    className="user-details"
  >
    <img
      alt="user avatar"
      className="avatar"
      src="https://avatars.githubusercontent.com/u/68243499?v=4"
    />
    <p>
      <a
        href="https://api.github.com/users/Juaneva"
      >
        https://api.github.com/users/Juaneva
      </a>
    </p>
    <h3>
      Latest 5 repositories:
    </h3>
    <div
      className="placeholder"
    >
      Loading...
    </div>
  </div>
  <div
    className="TBC"
  >
    <button
      className="back-button back-to-search-button"
      onClick={[Function]}
    >
      Go back to search results
    </button>
  </div>
</div>
`;
